spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesServiceInstance.java

```diff
@@ -16,211 +16,17 @@
 
 package org.springframework.cloud.kubernetes.commons.discovery;
 
-import java.net.URI;
-import java.util.Map;
-import java.util.Objects;
-
 import org.springframework.cloud.client.ServiceInstance;
-import org.springframework.core.style.ToStringCreator;
-
-import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryConstants.HTTP;
-import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryConstants.HTTPS;
-import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryConstants.NAMESPACE_METADATA_KEY;
-
-public final class KubernetesServiceInstance implements ServiceInstance {
```

Comment:
the idea I had was to take this class and transform it to a record, as such : `KubernetesAwareServiceInstance`. To make sure things stay the same I did this: I first wrote the unit tests and ran them against this class, then deleted it and re-ran them against the new record.